src/components/services/DeepLinkingManager.ts

```diff
@@ -1,104 +1,106 @@
+    (account !== defaultAccount && accountReferralLoaded && allWalletsLoaded && settingsLoaded) ||
+    // We can always handle recovery links:
+    pendingLink?.type === 'passwordRecovery'
 
-  // Startup tasks:
+  // Launches links, no matter how we got them:
   useAsyncEffect(
     async () => {
-      const listener = Linking.addEventListener('url', async event => await handleUrl(event.url))
-
-      let url = await Linking.getInitialURL()
-      if (url == null && ENV.YOLO_DEEP_LINK != null) url = ENV.YOLO_DEEP_LINK
-      if (url != null) await handleUrl(url)
-
-      return () => listener.remove()
+      if (!canHandleLink || pendingLink == null) return
+      setPendingLink(undefined)
+      await dispatch(launchDeepLink(navigation, pendingLink))
     },
-    [],
-    'DeepLinkingManager:getInitialURL'
+    [canHandleLink, dispatch, navigation, pendingLink],
+    'DeepLinkingManager:handleLink'
   )
 
-  const handlePushMessage = async (message: FirebaseMessagingTypes.RemoteMessage) => {
-    try {
-      const link = parsePushMessage(message)
-      if (link != null) {
-        await dispatch(launchDeepLink(navigation, link))
-      }
-    } catch (error) {
-      showError(error)
-    }
-  }
-
-  // Firebase messaging
+  // Subscribe to incoming links and load any app start-up links:
   useAsyncEffect(
     async () => {
-      /**
-       * Fires when the app launches from push notification
-       * */
-      const remoteMessage = await messaging().getInitialNotification()
-      if (remoteMessage != null) {
-        await handlePushMessage(remoteMessage)
+      function handleDeepLink(url: string): void {
+        try {
+          const link = parseDeepLink(url)
+          setPendingLink(link)
+        } catch (error: unknown) {
+          // The user tapped on the link, so show a real error:
+          showError(error)
+        }
       }
 
-      /**
-       * Fires when the app is in background
-       * */
-      messaging().onNotificationOpenedApp(remoteMessage => {
-        handlePushMessage(remoteMessage).catch(err => console.warn(err))
-      })
+      function handlePushMessage(message: FirebaseMessagingTypes.RemoteMessage): void {
+        try {
+          const link = parsePushMessage(message)
+          if (link != null) setPendingLink(link)
+        } catch (error: unknown) {
+          // The user does not initiate these, so use a dev error:
+          showDevError(error)
+        }
+      }
 
-      /**
-       * Fires when the app is in foreground and receives a notification
-       * */
-      const unsubscribe = messaging().onMessage(remoteMessage => {
-        // do nothing for now except return the unsubscribe function
+      // Subscribe to various incoming events:
+      const linkingCleanup = Linking.addEventListener('url', event => {
+        handleDeepLink(event.url)
+      })
+      const messageCleanup = messaging().onMessage(message => {
+        handlePushMessage(message)
```

Comment:
Should we actually handle`messaging().onMessag`? The previous code didn't

Reply:
This should be handled by showing the notification as a dropdown first mimicing the OS notification, and only if the user taps on it should we call handlePushNotification. Otherwise the user would automatically get routed to a buy/sell/swap modal in the middle of whatever they're doing.